### PR TITLE
Port vomit from Nyano

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Adds vomiting, available as a disease or reagent effect. Plus, new diseases and reagents to go along with it.

**Changelog**

:cl: Rane
- add: Ported vomiting from the Nyanotrasen fork. Several new and old diseases and chems now cause vomiting.
- fix: The vomit puddle sprite no longer loses its color as it evaporates.

### CODE REVIEW

File: Content.Server/Medical/VomitSystem.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                if (puddleSolution != null && _solutionSystem.TryGetSolution(stomach.Comp.Owner, StomachSystem.DefaultSolutionName, out var sol))
+                    _solutionSystem.TryAddSolution(puddle, puddleSolution, sol);
+            }
+            // And the small bit of the chem stream from earlier
+            if (TryComp<BloodstreamComponent>(uid, out var bloodStream))
+            {
+                var temp = bloodStream.ChemicalSolution.SplitSolution(solutionSize);
+                if (puddleSolution != null)
+                    _solutionSystem.TryAddSolution(puddle, puddleSolution, temp);
+            }
```
Comment (mirrorcult) [inline Content.Server/Medical/VomitSystem.cs:70]:
```suggestion
            if (!_solutionSystem.TryGetSolution(puddle, puddleComp.SolutionName, out var puddleSolution))
                return;

            // Empty the stomach out into it
            foreach (var stomach in stomachList)
            {
                if (_solutionSystem.TryGetSolution(stomach.Comp.Owner, StomachSystem.DefaultSolutionName, out var sol))
                    _solutionSystem.TryAddSolution(puddle, puddleSolution, sol);
            }
            // And the small bit of the chem stream from earlier
            if (TryComp<BloodstreamComponent>(uid, out var bloodStream))
            {
                var temp = bloodStream.ChemicalSolution.SplitSolution(solutionSize);
                _solutionSystem.TryAddSolution(puddle, puddleSolution, temp);
            }
```

File: Resources/Locale/en-US/disease/disease.ftl [hunk 2]
```diff
@@ -9,3 +9,4 @@ disease-beep= {CAPITALIZE($person)} beeps.
 disease-eaten-inside = You feel like you're being eaten from the inside.
 disease-banana-compulsion = You really want to eat some bananas.
 disease-beat-chest-compulsion = {CAPITALIZE(THE($person))} beats {POSS-ADJ($person)} chest.
+disease-vomit = {CAPITALIZE($person)} vomits.
```
Comment (mirrorcult) [inline Resources/Locale/en-US/disease/disease.ftl:12]:
```suggestion
disease-vomit = {CAPITALIZE(THE($person))} vomits.
```

File: Resources/Prototypes/Diseases/infectious.yml [hunk 3]
```diff
@@ -48,6 +48,51 @@
...
+      snoughSound:
+        collection: Coughs
+    - !type:DiseaseHealthChange
+      probability: 0.05
+      damage:
+        groups:
+          Caustic: 1
+  cures:
+    - !type:DiseaseBedrestCure
+      maxLength: 120
```
Comment (mirrorcult) [inline Resources/Prototypes/Diseases/infectious.yml:94]: this seems like basically the same thing as space flu, does it need to exist

**Review by mirrorcult — CHANGES_REQUESTED** [review]
there should be a special message if you try to vomit but there's nothing in your stomach imo

File: Resources/Prototypes/Diseases/infectious.yml (hunk 3 above)
Comment (Elijahrane) [reply Resources/Prototypes/Diseases/infectious.yml:94]: Having similar diseases helps to own the metagamers and require you to do at least a bit of differentiation

Comment (mirrorcult) [reply Resources/Prototypes/Diseases/infectious.yml:94]: sure why not

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**